tests/unit/testsuite/Bolt/Boltpay/Controller/Traits/WebHookTraitTest.php

```diff
@@ -0,0 +1,441 @@
+<?php
+
+require_once('TestHelper.php');
+require_once('StreamHelper.php');
+
+use Bolt_Boltpay_TestHelper as TestHelper;
+
+/**
+ * @coversDefaultClass Bolt_Boltpay_Controller_Traits_WebHookTrait
+ */
+class Bolt_Boltpay_Controller_Traits_WebHookTraitTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var string Dummy HMAC of test payload
+     * calculated as trim(base64_encode(hash_hmac('sha256', TEST_PAYLOAD, TEST_SIGNING_SECRET, true)))
+     */
+    const TEST_HMAC = 'fdd6zQftGT36/tGRItDZ0oB48VSptxj6TpZImLy4aZ4=';
+
+    /** @var string Dummy webhook payload */
+    const TEST_PAYLOAD = /** @lang JSON */ '{"test":"test"}';
+
+    /** @var string Dummy signing secret, same as md5('bolt') */
+    const TEST_SIGNING_SECRET = 'a6fe881cecd3fb7660083aea35cce430';
+
+    /**
+     * @var bool Used to check if fastcgi_finish_request is called, set to true by stubbed function
+     */
+    public static $_fastcgiFinishRequestCalled = false;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Bolt_Boltpay_Controller_Traits_WebHookTrait Mocked instance of the trait being tested
+     */
+    private $currentMock;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Mage_Core_Controller_Request_Http Mocked instance of request object
+     */
+    private $request;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Mage_Core_Controller_Response_Http Mocked instance of response object
+     */
+    private $response;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Mage_Core_Model_Layout Mocked instance of layout object
+     */
+    private $layout;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Bolt_Boltpay_Helper_Data Mocked instance of Bolt helper
+     */
+    private $helperMock;
+
+    /**
+     * Unregister helper set from previous tests
+     */
+    public static function setUpBeforeClass()
+    {
+        Mage::unregister('_helper/boltpay');
+    }
+
+    /**
+     * Setup test dependencies, called before each test
+     */
+    public function setUp()
+    {
+        Mage::app('default');
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Controller_Traits_WebHookTrait')
+            ->setMethods(array('getRequest', 'getLayout', 'getResponse'))
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->getMockForTrait();
+
+        $this->request = $this->getMockBuilder('Mage_Core_Controller_Request_Http')
+            ->setMethods(array('isAjax', 'getParam'))
+            ->getMock();
+
+        $this->layout = $this->getMockBuilder('Mage_Core_Model_Layout')
```

Comment:
The same as above

Reply:
A string is perfectly valid and even understood by IDEs like PHPStorm for code hinting.  The `::class` special constant, that also resolves to the exact same string, is only relevant when namespacing is in use. 

https://www.php.net/manual/en/language.oop5.constants.php

Namespacing is not in use in M1.